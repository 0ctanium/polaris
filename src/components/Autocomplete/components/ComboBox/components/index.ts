export {default as TextField} from './TextField';
export {Provider, Consumer} from './Context';
