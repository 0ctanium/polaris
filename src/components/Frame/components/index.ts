export {default as Toast, Props as ToastProps} from './Toast';
export {
  default as ToastManager,
  Props as ToastManagerProps,
} from './ToastManager';
export {default as Loading, Props as LoadingProps} from './Loading';
export {default as ContextBar, Props as ContextBarProps} from './ContextBar';
