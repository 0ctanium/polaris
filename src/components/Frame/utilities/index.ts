export {default as createFrameContext} from './createFrameContext';
