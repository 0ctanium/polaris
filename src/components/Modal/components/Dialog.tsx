import * as React from 'react';
import {classNames} from '@shopify/react-utilities/styles';
import {Transition, CSSTransition} from 'react-transition-group';
import {KeypressListener} from '../../';
import {Keys} from '../../../types';
import memoizedBind from '../../../utilities/memoized-bind';
import {TrapFocus} from '../../Focus';
import {Duration} from '../../shared';
import {AnimationProps} from '../../../types';
import * as styles from '../Modal.scss';

export interface DialogProps {
  labelledBy: string,
  instant?: boolean,
  children?: React.ReactNode,
  limitHeight?: boolean,
  large?: boolean,
  onClose(): void,
  onEntered?(): void,
  onExited?(): void,
}

export type Props = DialogProps & AnimationProps;

function DialogContainer(props: {children: React.ReactNode}) {
  return <div className={styles.Container}>{props.children}</div>;
}

export default function Dialog({
  instant,
  labelledBy,
  children,
  onClose,
  onExited,
  onEntered,
  large,
  limitHeight,
  ...props,
}: Props) {
  const classes = classNames(
    styles.Modal,
    large && styles['Modal-large'],
    limitHeight && styles['Modal-limitHeight'],
  );
  const handleClose = memoizedBind(onClose);
  const animation = large ? ScaleIn : FadeUp;
  const TransitionChild = instant ? Transition : animation;

  return (
    <TransitionChild
      {...props}
      data-polaris-layer
      data-polaris-overlay
      mountOnEnter
      unmountOnExit
      timeout={Duration.Base}
      onEntered={onEntered}
      onExited={onExited}
    >
      <DialogContainer>
        <TrapFocus>
          <div
            className={classes}
            role="dialog"
            aria-labelledby={labelledBy}
            tabIndex={-1}
          >
            <KeypressListener
              keyCode={Keys.ESCAPE}
              handler={handleClose}
              testID="CloseKeypressListener"
            />
            {children}
          </div>
        </TrapFocus>
      </DialogContainer>
    </TransitionChild>
  );
}

const fadeUpClasses = {
  enter: classNames(styles.FadeUp, styles.entering),
  enterActive: classNames(styles.FadeUp, styles.entered),
  exit: classNames(styles.FadeUp, styles.exiting),
  exitActive: classNames(styles.FadeUp, styles.exited),
};

function FadeUp({children, ...props}: any) {
  return (
    <CSSTransition {...props} classNames={fadeUpClasses}>
      {children}
    </CSSTransition>
  );
}

const scaleInClasses = {
  enter: classNames(styles.ScaleIn, styles.entering),
  enterActive: classNames(styles.ScaleIn, styles.entered),
  exit: classNames(styles.ScaleIn, styles.exiting),
  exitActive: classNames(styles.ScaleIn, styles.exited),
};

function ScaleIn({children, ...props}: any) {
  return (
    <CSSTransition {...props} classNames={scaleInClasses}>
      {children}
    </CSSTransition>
  );
}
