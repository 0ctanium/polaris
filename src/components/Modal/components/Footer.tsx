import * as React from 'react';
import {buttonsFrom, ButtonGroup, Stack} from '../../';
import {ComplexAction} from '../../../types';
import * as styles from '../Modal.scss';

export interface Props {
  primaryAction?: ComplexAction,
  secondaryActions?: ComplexAction[],
  children?: React.ReactNode,
}

export default function Footer({
  primaryAction,
  secondaryActions,
  children,
}: Props) {
  const primaryActionButton =
    (primaryAction && buttonsFrom(primaryAction, {primary: true})) || null;
  const secondaryActionButtons =
    (secondaryActions && buttonsFrom(secondaryActions)) || null;
  const actions =
    primaryActionButton || secondaryActionButtons ? (
      <ButtonGroup>
        {secondaryActionButtons}
        {primaryActionButton}
      </ButtonGroup>
    ) : null;

  return (
    <div className={styles.Footer}>
      <div className={styles.FooterContent}>
        <Stack alignment="center">
          <Stack.Item fill>{children}</Stack.Item>
          {actions}
        </Stack>
      </div>
    </div>
  );
}
