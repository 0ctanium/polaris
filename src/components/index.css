html {
  box-sizing: border-box;
  font-family: -apple-system, BlinkMacSystemFont, 'San Francisco', Roboto, 'Segoe UI', 'Helvetica Neue', sans-serif;
}

body {
  margin: 0;
  padding: 0;
  background-color: #ebeef0;
}

* {
  box-sizing: inherit;
}
