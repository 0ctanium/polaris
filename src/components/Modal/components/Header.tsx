import * as React from 'react';
import {Icon, DisplayText, Stack} from '@shopify/polaris';
import {memoizedBind} from '../../../utilities';
import * as styles from '../Modal.scss';

export interface Props {
  id: string;
  children?: React.ReactNode;
  onClose(): void;
}

export default function Header({id, children, onClose}: Props) {
  const handleClose = memoizedBind(onClose);

  return (
    <div className={styles.Header}>
      <Stack alignment="center">
        <Stack.Item fill>
          <div id={id}>
            <DisplayText element="h2" size="small">
              {children}
            </DisplayText>
          </div>
        </Stack.Item>

        <button onClick={handleClose} className={styles.CloseButton}>
          <Icon source="cancel" color="inkLighter" />
        </button>
      </Stack>
    </div>
  );
}
