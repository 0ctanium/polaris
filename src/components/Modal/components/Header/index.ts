import Header from './Header';

export {Props as HeaderProps} from './Header';
export default Header;

import CloseButton from './CloseButton';

export {CloseButton};
