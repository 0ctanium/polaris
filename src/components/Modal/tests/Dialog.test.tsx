import * as React from 'react';
import {mount} from 'enzyme';
import {KeypressListener} from '../../';

import {animationFrame, trigger} from '../../../../tests/utilities';

import {Dialog} from '../components';

describe('<Dialog>', () => {
  beforeEach(() => {
    animationFrame.fake();
  });

  afterEach(() => {
    animationFrame.restore();
  });

  it('renders CloseKeypressListener with correct props when `in` is true', () => {
    const listener = mount(
      <Dialog labelledBy="test" onClose={jest.fn()} in>
        something
      </Dialog>,
    ).find(KeypressListener);

    expect(listener.exists()).toBe(true);
    expect(listener.props()).toMatchSnapshot();
  });

  it('triggers an onEntered prop', () => {
    const dialog = mount(
      <Dialog labelledBy="test" onClose={jest.fn()} onEntered={jest.fn()}>
        something
      </Dialog>,
    );

    trigger(dialog, 'onEntered');
    expect(dialog.prop('onEntered')).toHaveBeenCalledTimes(1);
  });
});
