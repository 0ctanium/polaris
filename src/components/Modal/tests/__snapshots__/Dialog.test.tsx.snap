// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog> renders CloseKeypressListener with correct props when \`in\` is true 1`] = `
Object {
  "handler": [Function],
  "keyCode": 27,
  "testID": "CloseKeypressListener",
}
`;
